### PR TITLE
ExtraBuildEnv/ExtraArgs for qemu and tests

### PR DESCRIPTION
**Pull request by bluecmd**
u-bmc needs to pass a lot of parameters to qemu due to the nature of
ARM boards. Furthermore, u-bmc has a slightly more complex build system
than u-root and needs to pass some extra environment variables.

For ExtraBuildEnv nothing is using that particular value in u-root, but
as the same Options struct is passed around in u-bmc it would be nice to
add it to that structure.

Signed-off-by: Christian Svensson <bluecmd@google.com>

### CODE REVIEW

File: pkg/qemu/qemu.go [hunk 1]
```diff
@@ -57,6 +57,9 @@ type Options struct {
 
 	// Devices are devices to expose to the QEMU VM.
 	Devices []Device
+
+	// Extra parameters to pass to Qemu (used by u-bmc)
+	ExtraArgs []string
```
Comment (hugelgupf) [inline pkg/qemu/qemu.go]: use the qemu.ArbitraryArgs{} Device.

File: integration/integration.go [hunk 2]
```diff
@@ -91,6 +91,9 @@ type Options struct {
 
 	// Network is the VM's network.
 	Network *qemu.Network
+
+	// Extra environment variables to set when building (used by u-bmc)
+	ExtraBuildEnv []string
```
Comment (hugelgupf) [inline integration/integration.go]: Unused? Also, what's this for? Options.Env should have all Golang env variables you can set.

Comment (hugelgupf) [reply integration/integration.go]: Oh it's for make. Hm.

Comment (hugelgupf) [reply integration/integration.go]: I'm usually not a fan of broken abstractions like this, but I guess it's a test framework. Go for it?

**Comment by bluecmd** [comment]
I'll take a look at the Devices, that looks what I need.

As for the broken abstractions, I agree - but I'm not happy with the alternatives I tried. I'm happy to revisit if we think of something better down the line.

File: pkg/qemu/qemu.go (hunk 1 above)
Comment (bluecmd) [reply pkg/qemu/qemu.go]: Worked wonderfully, thanks!

**Comment by bluecmd** [comment]
PTAL @hugelgupf. It's only the ExtraBuildEnv thingy now.

**Pull request merged**